darts/models/regression_model.py

```diff
@@ -13,300 +13,259 @@
+
+        if model is None:
+            model = LinearRegression(n_jobs=-1, fit_intercept=False)
+
+        self.lags = lags
+        if isinstance(self.lags, int):
+            raise_if_not(self.lags > 0, "`lags` must be strictly positive. Given: {}.".format(self.lags))
+            self.lags = list(range(1, self.lags+1))
+        elif isinstance(self.lags, list):
+            for lag in self.lags:
+                raise_if(not isinstance(lag, int) or (lag <= 0), (
+                    "Every element of `lags` must be a strictly positive integer. Given: {}.".format(self.lags))
+                )
+
+        self.lags_exog = lags_exog
+        if self.lags_exog is True:
+            self.lags_exog = self.lags[:]
+        elif self.lags_exog is False:
+            self.lags_exog = [0]
+        elif isinstance(self.lags_exog, int):
+            raise_if_not(self.lags_exog > 0, "`lags_exog` must be strictly positive. Given: {}.".format(self.lags_exog))
+            self.lags_exog = list(range(1, self.lags_exog+1))
+        elif isinstance(self.lags_exog, list):
+            for lag in self.lags_exog:
+                raise_if(not isinstance(lag, int) or (lag < 0), (
+                    "Every element of `lags_exog` must be a positive integer. Given: {}."
+                    .format(self.lags_exog))
+                )
+
+        self.model = model
+        if self.lags is not None and self.lags_exog is None:
+            self.max_lag = int(np.max(self.lags))
+        elif self.lags is None and self.lags_exog is not None:
+            self.max_lag = int(np.max(self.lags_exog))
+        else:
+            self.max_lag = int(np.max([np.max(self.lags), np.max(self.lags_exog)]))
         self._fit_called = False
 
-    @abstractmethod
-    def fit(self, train_features: List[TimeSeries], train_target: TimeSeries) -> None:
+    def fit(self, series: TimeSeries, exog: Optional[TimeSeries] = None, **kwargs) -> None:
         """ Fits/trains the model using the provided list of features time series and the target time series.
 
         Parameters
         ----------
-        train_features
-            A list of features time series, all of the same length as the target series
-        train_target
-            A target time series, of the same length as the features series
+        series : TimeSeries
+            TimeSeries object containing the target values.
+        exog : TimeSeries, optional
+            TimeSeries object containing the exogenous values.
         """
+        super().fit(series, exog)
+        raise_if(exog is not None and self.lags_exog is None,
+            "`exog` not None in `fit()` method call, but `lags_exog` is None in constructor. "
+        )
+        raise_if(exog is None and self.lags_exog is not None,
+            "`exog` is None in `fit()` method call, but `lags_exog` is not None in constructor. "
+        )
+        self.target_column = series.columns()[0]
```

Comment:
(for later): could we extend this to work on all columns?
One idea could be, for the models that simply predict a scalar, to copy the model N times, and fit the N models to the N columns. Some models by be able to do vector regression, but not sure if there's a way to detect this somehow.

Reply:
Hm you mean regression on multiple different targets, not repetitions of the same time series, right?
I don't know which (scikit-learn) models support multiple regression but I think for now it's easier to leave the responsibility for that on the user-side. Otherwise we need to infer quite a lot from the input shape. Probably that's more complicated to deal with from our side.

Reply:
OK fair enough let's leave it like that for now.